tests/test_views.py

```diff
@@ -0,0 +1,1050 @@
+        )
+
+        assert response.context['project'] == project
+        assert len(response.context['url_list']) == 1
+        assert response.context['url_list'][0] == url
+
+
+class TestProjectUrls():
+
+    @pytest.mark.parametrize('request_type', [
+        'get',
+        'post'
+    ])
+    def test_status_ok(self, client, request_type):
+        project = factories.ProjectFactory()
+        response = getattr(client, request_type)(
+            reverse('project_urls', args=[project.project_slug]))
+        assert response.status_code == 200
+
+    @pytest.mark.parametrize('request_type', [
+        'get',
+        'post'
+    ])
+    def test_template_used(self, client, request_type):
+        project = factories.ProjectFactory()
+        response = getattr(client, request_type)(
+            reverse('project_urls', args=[project.project_slug]))
+        assert response.templates[0].name == 'nomination/project_urls.html'
+
+    @pytest.mark.parametrize('request_type', [
+        'get',
+        'post'
+    ])
+    def test_context(self, client, request_type):
+        project = factories.ProjectFactory()
+        factories.URLFactory.create_batch(
+            3,
+            url_project=project,
+            attribute='surt',
+            value='http://(com,example.www)'
+        )
+        factories.NominatedURLFactory.create_batch(
+            2,
+            url_project=project,
+            value=1
+        )
+        response = getattr(client, request_type)(
+            reverse('project_urls', args=[project.project_slug]))
+
+        assert response.context['project'] == project
+        assert response.context['url_count'] == 3
+        assert response.context['nominator_count'] == 2
+        for key, value in response.context['browse_tup']:
+            assert key == 'com'
+            for each in value:
+                if each[1] is not None:
+                    assert each == ('E', 'http://(com,e')
+
+
+class TestUrlListing():
+
+    @pytest.mark.parametrize('request_type, create_url', [
+        ('get', False),
+        ('get', True),
+        ('post', False),
+        ('post', True)
+    ])
+    def test_status_ok(self, client, request_type, create_url):
+        entity = 'www.example.com'
+        project = factories.ProjectFactory()
+        if create_url:
+            factories.URLFactory(url_project=project, entity=entity)
+        response = getattr(client, request_type)(
+            reverse('url_listing', args=[project.project_slug, entity]))
+
+        assert response.status_code == 200
+
+    @pytest.mark.parametrize('request_type, create_url, expected_template', [
+        ('get', False, 'nomination/url_add.html'),
+        ('post', False, 'nomination/url_add.html'),
+        ('get', True, 'nomination/url_listing.html'),
+        ('post', True, 'nomination/url_listing.html')
+    ])
+    def test_template_used(self, client, request_type, create_url, expected_template):
```

Comment:
Would you prefer the tests be split into two based on request type? They would have nearly identical bodies.